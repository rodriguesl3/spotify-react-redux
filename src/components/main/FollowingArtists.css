.card-react {
  background-color: #F6E8EA;
  margin: 20px 20px 20px 40px;
  padding: 20px;
  border-radius: 25px;
}

.card-img-top {
  width: 111%;
  border-top-left-radius: calc(.25rem - 1px);
  border-top-right-radius: calc(.25rem - 1px);
  position: absolute;
  left: -34px;
  border-radius: 25px;
  height: 160px;
}

.btn-success {
  background-color: #17DA19;
}
/*# sourceMappingURL=FollowingArtist.css.map */