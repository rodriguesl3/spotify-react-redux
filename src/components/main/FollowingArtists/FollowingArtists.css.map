{
    "version": 3,
    "mappings": "AAEA,AAAA,WAAW,CAAC;EACR,gBAAgB,ECMA,OAAO;EDLvB,MAAM,EAAE,mBAAmB;EAC3B,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,IAAI;CACtB;;AAED,AAAA,aAAa,CAAC;EACV,KAAK,EAAE,IAAI;EACX,sBAAsB,EAAE,kBAAkB;EAC1C,uBAAuB,EAAE,kBAAkB;EAC3C,QAAQ,EAAE,QAAQ;EAClB,IAAI,EAAE,KAAK;EACX,aAAa,EAAE,IAAI;EACnB,MAAM,EAAE,KAAK;CAChB;;AAED,AAAA,YAAY,CAAA;EACR,gBAAgB,ECZL,OAAO;CDarB",
    "sources": [
        "FollowingArtist.scss",
        "../../base.scss"
    ],
    "names": [],
    "file": "FollowingArtist.css"
}