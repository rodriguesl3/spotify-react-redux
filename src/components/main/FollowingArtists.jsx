import React from 'react';
// import PropTypes from 'prop-types';
import './FollowingArtists.scss';
import { GoThumbsup } from 'react-icons/go';
import { FaSpotify } from 'react-icons/fa';
import { RatingStart } from '../atoms/RatingStar';
import { CustomCard } from '../atoms/CustomCard';

const FollowingArtists = (props) => {
  const { followingArtist } = props;

  const renderCustomCard = item => (
    <div>
      <div className="row">
        <div className="col-12">
          <h5 className="card-title">{item.name}</h5>
          Popularity:
        </div>
        <div className="col-12">
          <RatingStart ratingValue={item.popularity} />
        </div>
      </div>
      <div className="row">
        <div className="card-text row">
          <p className="col-1">
            <GoThumbsup />
          </p>
          <p className="col-8">{item.followers.total}</p>
        </div>
        <button type="button" className="btn btn-success">
          More in
          <FaSpotify />
        </button>
      </div>
    </div>
  );

  return (
    <div>
      <h3 className="title">That is your idol list...</h3>
      {!followingArtist && <div>Loading</div>}
      <div className="row">
        {followingArtist
          && followingArtist.artists
            .items
            .map(item => (
              <CustomCard
                key={item.id}
                image={item.images[2].url}
                content={renderCustomCard(item)}
              />
            ))
        }
      </div>
    </div>
  );
};

export default FollowingArtists;
