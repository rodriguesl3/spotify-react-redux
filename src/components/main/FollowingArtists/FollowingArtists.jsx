import React from 'react';
// import PropTypes from 'prop-types';
import './FollowingArtists.scss';
import { GoThumbsup } from 'react-icons/go';
import { FaSpotify } from 'react-icons/fa';
import { RatingStart } from '../../atoms/RatingStar/RatingStar';
import { CustomCard } from '../../atoms/CustomCard/CustomCard';

import LoaderHOC from '../../atoms/Loading/LoadingHOC';


const FollowingArtists = (props) => {
  const { followingArtist } = props;

  const renderCustomCard = item => (
    <div>
      <div className="row">
        <div className="col-12">
          <h5 className="card-title">{item.name}</h5>
          Popularity:
        </div>
        <div className="col-12">
          <RatingStart ratingValue={item.popularity} />
        </div>
      </div>
      <div className="row">
        <div className="card-text row">
          <p className="col-1">
            <GoThumbsup />
          </p>
          <p className="col-8">{item.followers.total}</p>
        </div>
        <button type="button" className="btn btn-success">
          More in
          <FaSpotify />
        </button>
      </div>
    </div>
  );

  return (
    <div>
      <h3 className="title">That is your idol list...</h3>
      <div className="row">
        {followingArtist.artists
          .items
          .map(item => (
            <CustomCard
              key={item.id}
              image={item.images[2].url}
              content={renderCustomCard(item)}
            />
          ))
        }
      </div>
    </div>
  );
};

export default LoaderHOC('followingArtist', FollowingArtists);
