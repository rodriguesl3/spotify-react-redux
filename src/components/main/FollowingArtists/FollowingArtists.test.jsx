/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import FollowingArtists from './FollowingArtists';

test('FollowingArtists snaphot test', () => {
  const component = renderer.create(<FollowingArtists />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
