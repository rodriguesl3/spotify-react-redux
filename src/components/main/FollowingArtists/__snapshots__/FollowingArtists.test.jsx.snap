// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowingArtists snaphot test 1`] = `
<div>
  <h3
    className="title"
  >
    That is your idol list...
  </h3>
  <div>
    Loading
  </div>
  <div
    className="row"
  />
</div>
`;
